@here/harp-mapview/lib/MapView.ts

```diff
@@ -1610,24 +1682,23 @@ export class MapView extends THREE.EventDispatcher {
      * Returns the zoom level for the given camera setup.
      */
     get zoomLevel(): number {
-        return this.m_zoomLevel;
+        const cameraPitch = MapViewUtils.extractAttitude(this, this.m_camera).pitch;
+        const zoomLevelDistance =
+            this.m_lookAtDistance / Math.cos(Math.min(cameraPitch, Math.PI / 3));
+        return MapViewUtils.calculateZoomLevelFromDistance(zoomLevelDistance, this);
     }
+
     set zoomLevel(zoomLevel: number) {
-        this.m_zoomLevel = THREE.Math.clamp(zoomLevel, this.m_minZoomLevel, this.m_maxZoomLevel);
-        MapViewUtils.zoomOnTargetPosition(this, 0, 0, this.m_zoomLevel);
-        this.update();
+        THREE.Math.clamp(zoomLevel, this.m_minZoomLevel, this.m_maxZoomLevel);
+        this.lookAtDistance = MapViewUtils.calculateDistanceToGroundFromZoomLevel(this, zoomLevel);
     }
 
     /**
      * Returns the storage level for the given camera setup.
      * Actual storage level of the rendered data also depends on [[DataSource.storageLevelOffset]].
      */
     get storageLevel(): number {
-        return THREE.Math.clamp(
-            Math.floor(this.m_zoomLevel),
-            this.m_minZoomLevel,
-            this.m_maxZoomLevel
-        );
+        return this.zoomLevel;
```

Comment:
Why did you remove the floor? I think nobody is using this method but we should not break it's beheviour.